docs/src/references/mappers/auto_registration.md

```diff
@@ -0,0 +1,31 @@
+# Auto Registration of an entity
+
+Before any data messages from an entity can be processed, the entity has to be registered first.
+The entity can be registered either explicitly or implicitly (Auto registration).
+
+In `auto-registration`, the registration does not need a separate registration message, but the registration message will be derived from the first message that is sent by that entity.
+
+For example, sending a measurement message to te/device/child1///m/temperature, will result in the auto-registration of the entity with topic id: device/child1// and the auto-generated external id: <main-device-id>:device:child1, derived from the topic id. Similarly, a measurement message on te/device/child1/service/my-service/m/temperature, results in the auto-registration of both the device entity: device/child1// and the service entity: device/child1/service/my-service with their respective auto-generated external IDs, in that order.
+
+Pros:
+
+No need to have a separate registration message for an entity.
+
+Cons:
+
+When the message is sent for an nested child device or a service, then the registration will be done under the child device not under the nested child device.
+Because the mapper can’t understand is it for the child device or for the nested child device.
```

Comment:
```suggestion
**Pros:**

* No need to have a separate registration message for an entity.
   This would be ideal for simple devices where programming an additional registration logic is not possible ( e.g: simple sensors that can only generate telemetry messages).

**Cons:**

* Auto-registration of all entities is not possible in complex deployments with nested/hierarchical devices, as the parent of a nested child device can't be identified solely from its topic id (e.g: `te/device/nested-child//`).
The parent information must be provided explicitly using a registration message so that any nested child devices are not wrongly auto-registered as immediate child devices of the main device.
* Auto-registration results in the auto-generation of the device external id as well. If the user wants more control over it, then an explicit registration must be done.
```